### PR TITLE
Attempt to fix issues #776 and #652 by introducing a flag to disable near miss calculation and delay evaluation of request matching. 

### PR DESCRIPTION
**Pull request by nzCharlie**
This pull request have the following changes: 

- introduce a new flag into wiremock `Options` class `boolean reportNearMissDisabled()`. If the flag is on, we will not try to calculate near misses.
- delays evaluation of default matching in `RequestPattern`, additional matches are not evaluated after previous match fails (if the flag above is set, otherwise, matches are still evaluated when near miss calculation is performed)

### CODE REVIEW

**Comment by nzCharlie** [comment]
This pr should resolve issue #652 and issue #776

File: src/main/java/com/github/tomakehurst/wiremock/core/WireMockConfiguration.java [hunk 1]
```diff
@@ -85,6 +85,7 @@
 
     private Authenticator adminAuthenticator = new NoAuthenticator();
     private boolean requireHttpsForAdminApi = false;
+    private boolean reportNearMissDisabled = false;
```
Comment (AlixBa) [inline src/main/java/com/github/tomakehurst/wiremock/core/WireMockConfiguration.java]: I'm wondering if disabled by default isn't better

Comment (AlixBa) [reply src/main/java/com/github/tomakehurst/wiremock/core/WireMockConfiguration.java]: (I would also name this reportNearMiss as it's simpler to understand when you're setting its value)

**Comment by tomakehurst** [comment]
Let's back up here a little bit.

Disabling near miss calculation is the wrong thing to do. It doesn't (or shouldn't) happen during matching at all, only when you're asking for near misses so the problems this PR is meant to address won't be helped. If near miss calculation is taking place when it shouldn't, then this is the bug we should be fixing.

The excess of log messages can be fixed in the short term by removing them. The near miss feature provides a better way to figure out why your stub didn't match in any case.

A better way to do this might be to introduce a debug level on the notifier and switch the matcher messages to use this. That way you can keep info level logging enabled without flooding the logs with matcher messages.

**Comment by nzCharlie** [comment]
OK, in this case I will try to set the debug level of the warning message from matcher in question, and see if it will work for us.

**Pull request closed without merging**